botocore/docs/utils.py

```diff
@@ -124,3 +125,49 @@ def document_auto_populated_param(self, event_name, section, **kwargs):
             section = section.get_section('structure-value')
             if self.name in section.available_sections:
                 section.delete_section(self.name)
+
+
+class HideParamFromOperations(object):
+    """Hides a single parameter from multiple operations.
+
+    This method will remove a parameter from documentation and from
+    examples. This method is typically used for things that are
+    automatically populated because a user would be unable to provide
+    a value (e.g., a checksum of a serialized XML request body)."""
+    def __init__(self, service, parameter_name, operation_names):
+        """
+        :type parameter_name: str
+        :param parameter_name: Name of the parameter to modify.
+
+        :type operation_names: list
+        :param operation_names: Operation names to modify.
+        """
+        self._parameter_name = parameter_name
+        self._event_list = {}
+        # Build up a hash of relevant event names.
```

Comment:
I suggested it because the code looks a little cleaner if you use a list. Also you named it as ``event_list`` which implies it is a list. I do not think the list will take long to traverse as its size is limited to 2*(the number of parameters specified in the constructor).

Honestly, I am fine with either. I just suggested it because it took me a little long to understand what ``_event_list`` was representing. Maybe rename the variable to something more specific to what it represents like ``_event_type_dict``?